File: Models/BinderPageCardSlot.cs
using Microsoft.AspNetCore.Identity;

namespace CardCaptor.Models;

public class BinderPageCardSlot
{
    public int Id { get; set; }
    public int Position { get; set; }
    public int BinderPageId { get; set; }
    public BinderPage BinderPage { get; set; }
    public int? CardId { get; set; }
    public Card? Card { get; set; }
};

File: cardcaptor_app/lib/models/binder_page_card_slot.dart
import 'card.dart';

class BinderPageCardSlot {
  final int id;
  final int position;
  final int binderPageId;
  final int? cardId;
  final CardModel? card;

  BinderPageCardSlot({
    required this.id,
    required this.position,
    required this.binderPageId,
    this.cardId,
    this.card,
  });

  factory BinderPageCardSlot.fromJson(Map<String, dynamic> json) {
    return BinderPageCardSlot(
      id: json['id'],
      position: json['position'],
      binderPageId: json['binderPageId'],
      cardId: json['cardId'],
      card: json['card'] != null ? CardModel.fromJson(json['card']) : null,
    );
  }
}
